packages/edit-site/src/components/sidebar-edit-mode/index.js

```diff
@@ -22,27 +25,77 @@ import TemplatePanel from './template-panel';
 import PluginTemplateSettingPanel from '../plugin-template-setting-panel';
 import { SIDEBAR_BLOCK, SIDEBAR_TEMPLATE } from './constants';
 import { store as editSiteStore } from '../../store';
+import { unlock } from '../../lock-unlock';
+
+const { Tabs } = unlock( componentsPrivateApis );
 
 const { Slot: InspectorSlot, Fill: InspectorFill } = createSlotFill(
 	'EditSiteSidebarInspector'
 );
 export const SidebarInspectorFill = InspectorFill;
 
+const FillContents = ( {
+	sidebarName,
+	isEditingPage,
+	supportsGlobalStyles,
+} ) => {
+	// Because `DefaultSidebar` renders a `ComplementaryArea`, we
+	// need to forward the `Tabs` context so it can be passed through the
+	// underlying slot/fill.
+	const tabsContextValue = useContext( Tabs.Context );
+
+	return (
+		<>
+			<DefaultSidebar
+				identifier={ sidebarName }
+				title={ __( 'Settings' ) }
+				icon={ isRTL() ? drawerLeft : drawerRight }
+				closeLabel={ __( 'Close Settings' ) }
+				header={
+					<Tabs.Context.Provider value={ tabsContextValue }>
+						<SettingsHeader />
+					</Tabs.Context.Provider>
+				}
+				headerClassName="edit-site-sidebar-edit-mode__panel-tabs"
+				// This classname is added so we can apply a corrective negative
+				// margin to the panel.
+				// see https://github.com/WordPress/gutenberg/pull/55360#pullrequestreview-1737671049
+				className="edit-site-sidebar__panel"
+			>
+				<Tabs.Context.Provider value={ tabsContextValue }>
+					<Tabs.TabPanel id={ SIDEBAR_BLOCK } focusable={ false }>
+						{ isEditingPage ? <PagePanels /> : <TemplatePanel /> }
+						<PluginTemplateSettingPanel.Slot />
+					</Tabs.TabPanel>
+
+					<Tabs.TabPanel id={ SIDEBAR_TEMPLATE } focusable={ false }>
+						<InspectorSlot bubblesVirtually />
+					</Tabs.TabPanel>
```

Comment:
By comparing the code with the one on trunk, it looks like the tabpanel IDs are swapped?

```suggestion
					<Tabs.TabPanel id={ SIDEBAR_TEMPLATE } focusable={ false }>
						{ isEditingPage ? <PagePanels /> : <TemplatePanel /> }
						<PluginTemplateSettingPanel.Slot />
					</Tabs.TabPanel>

					<Tabs.TabPanel id={ SIDEBAR_BLOCK } focusable={ false }>
						<InspectorSlot bubblesVirtually />
					</Tabs.TabPanel>
```

Reply:
Right you are. And there's more here as well, actually. Reversing the IDs should effectively reverse the tabpanel contents as well, but it doesn't with the current changes. That's why I didn't spot this prior to you pointing it out.

I mistakenly (stupid muscle memory) used `id` instead of `tabId` and without TS on this file to yell at me I didn't notice it until now.

That means the tabpanels aren't looking to be controlled by a specific tab at all, they're just getting linked under the hood based on the order they appear in the code. The fact that it "works" in the UI this way is just a lucky coincidence.

I'm going to fix the reversed IDs you've pointed out, pass them as proper `tabId` props, and then when #57033 merges, I'll rebase this PR and confirm that they behave as expected.

Reply:
Done! 3d9c1dcb9e

Thanks again for spotting that!